File: Data-Structure/Assignment_39_5.java
import java.util.*;

class Node
{
   public int data;
   public Node next;
   
   public Node(int Value)
   {
      this.data = Value;
      this.next = null;
   }
}

class SinglyLinkedList
{
   private Node first;
   
   public SinglyLinkedList()
   {
      this.first = null;
   }
   
   public void InsertLast(int iNo)
   {
      Node newn = new Node(iNo);
      if(first == null)
      {
         first = newn;
      }
      else
      {
         Node temp = first;
         while(temp.next != null)
         {
            temp = temp.next;
         }
         temp.next = newn;
      }
   }

   public void Display()
   {
      Node temp = first;
      while(temp != null)
      {
         System.out.print("| " + temp.data + " | -> ");
         temp = temp.next;
      }
      System.out.println("NULL");
   }

   public void DisplayEven()
   {
      Node temp = first;
      while(temp != null)
      {
         if(temp.data % 2 == 0)
         {
            System.out.print("| " + temp.data + " | -> ");
         }
         temp = temp.next;
      }
      System.out.println("NULL");
   }
}

class Assignment_39_5
{
   public static void main(String A[])
   {
      SinglyLinkedList ssobj = new SinglyLinkedList();

      ssobj.InsertLast(21);
      ssobj.InsertLast(10);
      ssobj.InsertLast(51);
      ssobj.InsertLast(102);
      ssobj.InsertLast(111);

      ssobj.Display();

      ssobj.DisplayEven();
   }
}

File: C-Programming/Competitive/Assignment_39_5.c
#include <stdio.h>
#include <stdlib.h>

struct node
{
   int data;
   struct node *next;
};

typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

void DisplayEven(PNODE first)
{
   while (first != NULL)
   {
      if(first->data % 2 == 0)
      {
         printf("%d\n", first->data);
      }
      first = first->next;
   }
}

void InsertFirst(PPNODE first, int iNo)
{
   PNODE newn = NULL;

   newn = (PNODE)malloc(sizeof(NODE));

   newn->data = iNo;
   newn->next = NULL;

   if(*first == NULL)
   {
      *first = newn;
   }
   else
   {
      newn->next = *first;
      *first = newn;
   }
}

int main()
{
   PNODE head = NULL;

   InsertFirst(&head,20);
   InsertFirst(&head,11);
   InsertFirst(&head,54);
   InsertFirst(&head,21);
   InsertFirst(&head,16);

   DisplayEven(head);

   return 0;
}